### ISSUE TITLE
Support for incremental deployment

### ISSUE THREAD

**Issue by daveaglick** [statiqdev/Statiq#991]
There is an unserved use case where only parts of the generation are run and the resulting output files only represent a subset of all files that constitute a full site. When running in this mode we need a way to deploy only those changed files without deleting everything else that's already been deployed.

Right now the built-in deployment modules generally upload and deploy everything in a "full merge" mode. In GitHub this means uploading every output document as a blob and creating a full commit tree. In Azure and Netlify this means uploading a zip file with the full output. Ideally we'd have a setting for every deployment environment that toggles from a "full merge" mode to a "overwrite changes" mode - though that's highly dependent on what the various environments support.

cc @Simply007

**Comment by atiq-cs** [comment]
Highly desirable feature indeed!

**Comment by ghost** [comment]
i agree @atiq-cs, perhaps this also affects the current issue where building with github pages that build from a specific branch will wipe out the existing CNAME file [when using a custom domain, specific to branch based pages deployment].
